commit-graph.c

```diff
@@ -747,7 +765,18 @@ static void fill_commit_graph_info(struct commit *item, struct commit_graph *g,
 	date_low = get_be32(commit_data + g->hash_len + 12);
 	item->date = (timestamp_t)((date_high << 32) | date_low);
 
-	graph_data->generation = get_be32(commit_data + g->hash_len + 8) >> 2;
+	if (g->chunk_generation_data) {
+		offset = (timestamp_t) get_be32(g->chunk_generation_data + sizeof(uint32_t) * lex_index);
+
+		if (offset & CORRECTED_COMMIT_DATE_OFFSET_OVERFLOW) {
+			offset_pos = offset - CORRECTED_COMMIT_DATE_OFFSET_OVERFLOW;
```

Comment:
I wonder if this would be clearer using bitwise XOR instead of subtraction:

`offset_pos = offset ^ CORRECTED_COMMIT_DATE_OFFSET_OVERFLOW;`